tests/unit/test_tf_utils.py

```diff
@@ -0,0 +1,148 @@
+    assert type(model) == tf.estimator.LinearRegressor
+    # Test if wide column has one cross-product column
+    assert len(wide_columns) == 1
+    assert len(deep_columns) == 0
+    # Test if model train works
+    model.train(
+        input_fn=pandas_input_fn(df=data, y_col=DEFAULT_RATING_COL, batch_size=1, num_epochs=10, shuffle=True)
+    )
+    _clean_up('wide_'+MODEL_DIR)
+
+    model, wide_columns, deep_columns = build_model(users, items, model_dir='deep_'+MODEL_DIR, model_type='deep')
+    assert type(model) == tf.estimator.DNNRegressor
+    # Test if deep columns have user and item columns
+    assert len(wide_columns) == 0
+    assert len(deep_columns) == 2
+    # Test if model train works
+    model.train(
+        input_fn=pandas_input_fn(df=data, y_col=DEFAULT_RATING_COL, batch_size=1, num_epochs=10, shuffle=True)
+    )
+    _clean_up('deep_'+MODEL_DIR)
+
+    model, wide_columns, deep_columns = build_model(users, items, model_dir='wide_deep_'+MODEL_DIR, model_type='wide_deep')
+    assert type(model) == tf.estimator.DNNLinearCombinedRegressor
+    assert len(wide_columns) == 1
+    assert len(deep_columns) == 2
+    # Test if model train works
+    model.train(
+        input_fn=pandas_input_fn(df=data, y_col=DEFAULT_RATING_COL, batch_size=1, num_epochs=10, shuffle=True)
+    )
+    _clean_up('wide_deep_'+MODEL_DIR)
+
+
+def test_evaluation_log_hook(pd_df):
+    data, users, items = pd_df
+
+    # Run hook 10 times
+    hook_frequency = 10
+    train_steps = 100
+
+    model, wide_columns, deep_columns = build_model(
+        users, items, model_dir='deep_'+MODEL_DIR, model_type='deep', save_checkpoints_steps=train_steps//hook_frequency
+    )
+
+    class EvaluationLogger(Logger):
+        def __init__(self):
+            self.eval_log = {}
+
+        def log(self, metric, value):
+            if metric not in self.eval_log:
+                self.eval_log[metric] = []
+            self.eval_log[metric].append(value)
+
+        def get_log(self):
+            return self.eval_log
+
+    evaluation_logger = EvaluationLogger()
+
+    hooks = [
+        evaluation_log_hook(
+            model,
+            logger=evaluation_logger,
+            true_df=data,
+            y_col=DEFAULT_RATING_COL,
+            eval_df=data.drop(DEFAULT_RATING_COL, axis=1),
+            every_n_iter=train_steps//hook_frequency,
+            model_dir='deep_'+MODEL_DIR,
+            eval_fns=[rmse],
+        )
+    ]
+    model.train(
+        input_fn=pandas_input_fn(df=data, y_col=DEFAULT_RATING_COL, batch_size=1, num_epochs=None, shuffle=True),
+        hooks=hooks,
+        steps=train_steps
+    )
+    _clean_up('deep_' + MODEL_DIR)
+
+    # Check if hook logged the given metric
+    assert rmse.__name__ in evaluation_logger.get_log()
+    assert len(evaluation_logger.get_log()[rmse.__name__]) == hook_frequency
+
+
+def _clean_up(path):
```

Comment:
maybe we can move this to a general files, this could be used by other algos. Also, we can try to find a more descriptive name, remove_folders, remove_path...

Reply:
oh. that function is not needed anymore. I changed all to directly call `shutil.rmtree` in the next commit.